### PR TITLE
chore: Update datafusion

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
1. Keep up with dependencies
2. Closes https://github.com/influxdata/influxdb_iox/issues/8264 (with https://github.com/apache/arrow-datafusion/pull/7124)

# Changes

1. Update to DataFusion pin to https://github.com/apache/arrow-datafusion/commit/2cf5f5b5bb824598de185d64c541c52c930728cf

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1530,28 +1529,17 @@ dependencies = [
  "prost",
 ]
 
-[[package]]
-name = "datafusion-row"
```
Comment (alamb) [inline Cargo.lock:1534]: this is some other fallout from the datafusion grouping work -- removed an old style row thingie

File: iox_query/src/logical_optimizer/handle_gapfill.rs [hunk 2]
```diff
@@ -198,7 +198,9 @@ fn build_gapfill_node(
     let time_column =
         col(new_aggr_plan.schema().fields()[date_bin_gapfill_index].qualified_column());
 
-    let aggr = Aggregate::try_from_plan(&new_aggr_plan)?;
```
Comment (alamb) [inline iox_query/src/logical_optimizer/handle_gapfill.rs:201]: was removed in https://github.com/apache/arrow-datafusion/pull/7069

**Comment by alamb** [comment]
This PR is hitting a panic in datafusion code. Likely introduced in https://github.com/apache/arrow-datafusion/pull/6982

To reproduce

```shell
RUST_LOG=debug LOG_FILTER=debug RUST_BACKTRACE=1 CARGO_TARGET_DIR=/Users/alamb/Software/target-iox2 TEST_INFLUXDB_JDBC=true TEST_INTEGRATION=1 TEST_INFLUXDB_IOX_CATALOG_DSN=postgresql://postgres@localhost:5432/postgres cargo test --test end_to_end -- issue_6112
```

```
2023-07-27T17:36:37.278019Z DEBUG datafusion::datasource::physical_plan::parquet: Creating ParquetExec, files: [[PartitionedFile { object_meta: ObjectMeta { location: Path { raw: "1/1/4710107baa934df3ee81038c93840af52a74c16e2c4203154b678aec3d59429f/b6231d65-66db-499f-90d1-68ee4ae71d50.parquet" }, last_modified: 1970-01-01T00:00:00Z, size: 3286, e_tag: None }, partition_values: [Int64(1690479396848478000)], range: None, extensions: Some(Any { .. }) }]], projection None, predicate: None, limit: None    
2023-07-27T17:36:37.278211Z DEBUG datafusion::datasource::physical_plan::parquet: Creating ParquetExec, files: [[PartitionedFile { object_meta: ObjectMeta { location: Path { raw: "1/1/4710107baa934df3ee81038c93840af52a74c16e2c4203154b678aec3d59429f/b6231d65-66db-499f-90d1-68ee4ae71d50.parquet" }, last_modified: 1970-01-01T00:00:00Z, size: 3286, e_tag: None }, partition_values: [Int64(1690479396848478000)], range: None, extensions: Some(Any { .. }) }]], projection None, predicate: None, limit: None    
2023-07-27T17:36:37.278391Z DEBUG datafusion::datasource::physical_plan::parquet: Creating ParquetExec, files: [[PartitionedFile { object_meta: ObjectMeta { location: Path { raw: "1/1/4710107baa934df3ee81038c93840af52a74c16e2c4203154b678aec3d59429f/b6231d65-66db-499f-90d1-68ee4ae71d50.parquet" }, last_modified: 1970-01-01T00:00:00Z, size: 3286, e_tag: None }, partition_values: [Int64(1690479396848478000)], range: None, extensions: Some(Any { .. }) }]], projection None, predicate: Some(BinaryExpr { left: BinaryExpr { left: Column { name: "i64", index: 1 }, op: BitwiseAnd, right: Literal { value: Int64(1) } }, op: Eq, right: Literal { value: Int64(0) } }), limit: None    
2023-07-27T17:36:37.278637Z DEBUG datafusion::datasource::physical_plan::parquet: Creating ParquetExec, files: [[PartitionedFile { object_meta: ObjectMeta { location: Path { raw: "1/1/4710107baa934df3ee81038c93840af52a74c16e2c4203154b678aec3d59429f/b6231d65-66db-499f-90d1-68ee4ae71d50.parquet" }, last_modified: 1970-01-01T00:00:00Z, size: 3286, e_tag: None }, partition_values: [Int64(1690479396848478000)], range: None, extensions: Some(Any { .. }) }]], projection Some([1, 5]), predicate: Some(BinaryExpr { left: BinaryExpr { left: Column { name: "i64", index: 1 }, op: BitwiseAnd, right: Literal { value: Int64(1) } }, op: Eq, right: Literal { value: Int64(0) } }), limit: None    
2023-07-27T17:36:37.278979Z ERROR panic_logging: Thread panic panic_type="unknown" panic_info=panicked at 'internal error: entered unreachable code', /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:211:14
thread 'IOx Query 1' panicked at 'internal error: entered unreachable code', /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:211:14
stack backtrace:
   0: rust_begin_unwind
             at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:67:14
   2: core::panicking::panic
             at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/panicking.rs:117:5
   3: datafusion_physical_expr::intervals::cp_solver::get_inverse_op
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:211:14
   4: datafusion_physical_expr::intervals::cp_solver::propagate_arithmetic
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:233:22
   5: <datafusion_physical_expr::expressions::binary::BinaryExpr as datafusion_physical_expr::physical_expr::PhysicalExpr>::propagate_constraints
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/expressions/binary.rs:794:13
   6: datafusion_physical_expr::intervals::cp_solver::ExprIntervalGraph::propagate_constraints
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:505:40
   7: datafusion_physical_expr::intervals::cp_solver::ExprIntervalGraph::update_ranges
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/91d7806/datafusion/physical-expr/src/intervals/cp_solver.rs:531:26
   8: datafusion_physical_expr::physical_expr::analyze

```

I will work on an upstream reproducer tomororw

**Comment by alamb** [comment]
Upstream issues: https://github.com/apache/arrow-datafusion/issues/7125

**Review by crepererum — APPROVED** [review]

**Pull request merged**